### DISCUSSION TITLE
Watched content media management tool

### DISCUSSION THREAD

**Discussion by marcodejongh** [sct/overseerr#3436] [Feature Requests]
Some media server have relatively limited disk space, but do have a ton of bandwidth.
Meaning, that those servers would like to optimise for short retention but frequent downloading, rather than data hoarding.
Overseerr is key to making that usecase work too as it makes it easy for users to request new stuff, and with the sync rework re-requesting previously downloaded content should become possible too: https://github.com/sct/overseerr/pull/3219.

I think ideally this would be a feature that should go into Overseerr too, I've seen: https://github.com/sct/overseerr/discussions/2335#discussioncomment-1779557
Saying Overseerr should not delete media files through it's interface.
And I agree Overseerr shouldn't touch files directly, but does this statement also extend to hitting the file management API's of *Arr apps?

The reason why I would argue this is Overseerr functionality, is that Overseerr owns the requests. So Overseerr is the tool that has knowledge of what users has requested what media. It could then use the API's of Plex or Tautulli to determine what requests have been watched and those can be flagged for deletion. I'm personally not too worried about Overseerr managing deletions since *Arr apps have a recycling bin system, so erroneously deleted data should be easy to retrieve when *Arr is configured properly. To reduce worries of accidental deletion, we could possible require recycling bin in *Arr to have been configured as a hard requirement for this new media management feature to work.

There's definitely a lot more considerations to this feature. For example, if user A requested TV Show X and has watched it, but user B has also started watching it, we probably would want to surface that through the UI too. But initially, we could just start out with only showing the easy removal decisions where user A has requested it and finished watching and nobody also has watched it.

If the maintainers want to be firm on not deleting media, a compromise would be to have a rest-api that can easily surface this information to make it a lot easier for me to implement a CLI that can do media cleanup. Or Overseerr could tag items in *Arr as watched by requesters

For this workflow to work properly allowing multiple users to request the same thing would also be helpful: https://github.com/sct/overseerr/discussions/2687

**Comment by Apollo5555** [comment]
I agree and would like to see this as well.
For me, just a simple report of requested shows, that have been watched, that I can then manually remove would be very helpful.
I don't necessary want to keep odd shows that people request after they have watched them.

**Comment by justkorix** [comment]
I would also love to add a restriction of requests if they are requesting but not watching. So it would set a limit of say 5 movies or seasons until they watch some of them to open up more request slots
